rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/ReleaseController.java

```diff
@@ -270,12 +298,314 @@ public void downloadAttachmentFromRelease(
+    public ResponseEntity<HalResource> triggerFossologyProcess(@PathVariable("id") String releaseId,
+            @RequestParam(value = "markFossologyProcessOutdated", required = false) boolean markFossologyProcessOutdated,
+            HttpServletResponse response) throws TException, IOException {
+        releaseService.checkFossologyConnection();
+
+        ReentrantLock lock = mapOfLocks.get(releaseId);
+        Map<String, String> responseMap = new HashMap<>();
+        HttpStatus status = null;
+        if (lock == null || !lock.isLocked()) {
+            if (mapOfLocks.size() > 10) {
+                responseMap.put("message",
+                        "Max 10 FOSSology Process can be triggered simultaneously. Please try after sometime.");
+                status = HttpStatus.TOO_MANY_REQUESTS;
+            } else {
+                executeFossologyProcess(releaseId, markFossologyProcessOutdated);
+                responseMap.put("message", "FOSSology Process for Release Id : " + releaseId + " has been triggered.");
+                status = HttpStatus.OK;
+            }
+
+        } else {
+            status = HttpStatus.NOT_ACCEPTABLE;
+            responseMap.put("message", "FOSSology Process for Release Id : " + releaseId
+                    + " is already running. Please wait till it is completed.");
+        }
+        HalResource responseResource = new HalResource(responseMap);
+        Link checkStatusLink = linkTo(ReleaseController.class)
+                .slash("api/releases/" + releaseId + "/checkFossologyProcessStatus").withSelfRel();
+        responseResource.add(checkStatusLink);
+
+        return new ResponseEntity<HalResource>(responseResource, status);
+    }
+
     @Override
     public RepositoryLinksResource process(RepositoryLinksResource resource) {
         resource.add(linkTo(ReleaseController.class).slash("api" + RELEASES_URL).withRel("releases"));
         return resource;
     }
 
+    private void executeFossologyProcess(String releaseId, boolean markFossologyProcessOutdated)
+            throws TException, IOException {
+        User user = restControllerHelper.getSw360UserFromAuthentication();
+        String attachmentId = validateNumberOfSrcAttached(releaseId, user);
+
+        if (markFossologyProcessOutdated) {
+            log.info("Marking FOSSology process outdated for Release : " + releaseId);
+            releaseService.markFossologyProcessOutdated(releaseId, user);
+        }
+
+        Release release = releaseService.getReleaseForUserById(releaseId, user);
+
+        ExternalToolProcess fossologyProcess = releaseService.getFossologyProcess(release);
+        if (fossologyProcess != null && isFOSSologyProcessCompleted(fossologyProcess)) {
+            log.info("FOSSology process for Release : " + releaseId + " already completed.");
+            return;
+        }
+
+        final ExternalToolProcess fossologyProcessFinal = fossologyProcess;
```

Comment:
`fossologyProcessFinal` is used in lambda expression, which needs to be final or implicitly final.
`fossologyProcess` cannot be used inside Lambda Exp as its value changes over the control flow.
Please suggest if this can be achieved by any other means.